### PR TITLE
2hx96yu generator UI redesign

### PR DESCRIPTION
**Pull request by shubham-shay-te**

### CODE REVIEW

File: website/src/components/MenuTitleChildren/MenuTitleChildren.jsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                
+            </div>
+        
+    )
+}
+
+const MenuTitleChildren = (props) => {
+    const children = []
+    useEffect(()=>{}, [props.children])
+    props.children.forEach((child, index) => {
```
Comment (shay-te) [inline website/src/components/MenuTitleChildren/MenuTitleChildren.jsx:41]: in action, `props.children` is any content passed in the body. any inner component. 
if we pass to it an array of JSON items. it's better to dedicate a name to it. like `items`

File: website/src/components/MenuTitleChildren/MenuTitleChildren.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        children.push(
+           <RenderChild key={index} path={child.path} name={child.name} index={index} onClick={props.onClick} onDeleteClick={props.onDeleteClick} showDelete={props.showDelete}/>
+        )
+    })
+
+    let renderChildren = ''
+    if(props.showChildren){
+        renderChildren = (children.length > 0 ? children : <div className={`child child-info`}>No entries in this section.</div>)
```
Comment (shay-te) [inline website/src/components/MenuTitleChildren/MenuTitleChildren.jsx:49]: if children are passed from props. so messages that describe this message should.
we can add a property called `noEntriesMessage`

That way we can also replace the message and reuse this components in many locations.

File: website/src/components/inputTable/ColumnsTable.jsx [hunk 3]
```diff
@@ -1,65 +1,87 @@
...
+            setColumns(props.value)
+        }
+        
+    }, [props.value.length])
     items.push(
-        props.value.map((column, index) => {
+        columns.map((column, index) => {
+            let validatorCallback;
+            if(column.type === 'INTEGER'){
+                validatorCallback = isNumber
```
Comment (shay-te) [inline website/src/components/inputTable/ColumnsTable.jsx:22]: is this logic need to be inside the fields generator? can we make a function that gets a field type and return the correct validation? and use it in the field generator.

File: website/src/components/inputTable/ColumnsTable.jsx [hunk 4]
```diff
@@ -1,65 +1,87 @@
...
     const [visible, setVisible] = useState(false);
-
+    const [columns, setColumns] = useState([]);
     const items = [];
+    useEffect(() => {
+        if(columns.length !== props.value.length){
+            setColumns(props.value)
+        }
+        
+    }, [props.value.length])
```
Comment (shay-te) [inline website/src/components/inputTable/ColumnsTable.jsx:17]: so we don't with to capture change inside a specific field. only if fields were removed or added?

File: website/src/components/inputTable/FunctionsTable.jsx [hunk 5]
```diff
@@ -1,35 +1,45 @@
...
+    useEffect(() => {
+        if(functions.length !== props.value.length){
+            setFunctions(props.value)
+        }
         
-        props.value.map((func, index) => {
-            
+    }, [props.value.length])
+    items.push(
+        functions.map((func, index) => {
```
Comment (shay-te) [inline website/src/components/inputTable/FunctionsTable.jsx:19]: `functions.map` will return an array. and items are already an array.

can we just assign `items  = functions.map((fun....` ?

File: website/src/fieldsGenerator/downloadZipFields.js [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+export const downloadZipFields = () => {
+    const fields = [];
+    
+    fields.push(
+        {
+            title: 'Download Core-Lib zip file',
+            subtitle: 'Will generate and download a new Core-Lib zip file as configured in the generator. ' +
+                      'After you have the zip file with you, you can then place the Core-Lib inside your project and start using it without any hassle.',
+            type: 'title_subtitle',
+            // validatorCallback: validateFunc,
```
Comment (shay-te) [inline website/src/fieldsGenerator/downloadZipFields.js:12]: no remarks. place validation only if needed

File: website/src/fieldsGenerator/exportYamlFields.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    const fields = [];
+    fields.push(
+        {
+            title: 'Export YAML file',
+            subtitle: 'Will generate and download a new YAML file with the configured Core-Lib generator config. ' +
+                      'This YAML contains the data that will help the generator generate a new Core-Lib for you. ' +
+                      'You can either select the Download Zip option and download the zip file of your Core-Lib right away or '+
+                      'you can use the command `core_lib -g YourCoreLib.yaml` to generate a new Core-Lib from the command line.',
+            type: 'title_subtitle',
+            // validatorCallback: validateFunc,
```
Comment (shay-te) [inline website/src/fieldsGenerator/exportYamlFields.js:13]: remove validation

File: website/src/utils/YamlData.js [hunk 8]
```diff
@@ -130,14 +130,26 @@ export class YamlData {
...
         if(!list){
             return res
         }
         if (path === 'core_lib.entities') {
             const entityRes = []
-            list.forEach((entity, index) => {
-                entityRes.push({ name: `${entity.key} (${entity.connection})`, path: path + '.' + index, dbConnection: entity.connection })
+            const connectionList = this.get('core_lib.connections')
+            connectionList.forEach((connection) => {
+                if(connection.type.includes('SqlAlchemyConnectionRegistry')) {
```
Comment (shay-te) [inline website/src/utils/YamlData.js:141]: please explain

File: website/src/utils/validatorUtils.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
+export const isPascalCase = (string) => {
```
Comment (shay-te) [inline website/src/utils/validatorUtils.js:1]: Good work!

**Review by shay-te — CHANGES_REQUESTED** [review]

File: website/src/components/MenuTitleChildren/MenuTitleChildren.jsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+                        </HoverVisible>
+                    : ''
+                }
+                
+            </div>
+        
+    )
+}
+
+const MenuTitleChildren = (props) => {
```
[the remark this answers is not in the document]
Comment (shubham-shay-te) [reply website/src/components/MenuTitleChildren/MenuTitleChildren.jsx:38]: the other compoenent is a local component that was specifically made for this component to handle the child events so thought of keep it locally

File: website/src/components/inputTable/ColumnsTable.jsx (hunk 4 above)
Comment (shubham-shay-te) [reply website/src/components/inputTable/ColumnsTable.jsx:17]: yes, just to update the state that holds these values

File: website/src/utils/validatorUtils.js (hunk 9 above)
Comment (shubham-shay-te) [reply website/src/utils/validatorUtils.js:1]: thankyou

**Review by shay-te — APPROVED** [review]

**Pull request merged**